modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc/JdbcStatement.java

```diff
@@ -711,10 +708,37 @@ void ensureNotClosed() throws SQLException {
      *
      * @throws SQLException On error.
      */
-    protected void closeResults() throws SQLException {
+    void closeResults() throws SQLException {
+        @Nullable JdbcResultSet last = null;
+
         if (resSets != null) {
-            for (JdbcResultSet rs : resSets) {
-                rs.close0();
+            JdbcResultSet lastRs = resSets.get(resSets.size() - 1);
+            boolean allFetched = lastRs == null;
+
+            if (allFetched) {
+                for (JdbcResultSet rs : resSets) {
+                    if (rs != null) {
+                        rs.close0(true);
+                    }
+                }
+            } else {
+                try {
+                    last = lastRs.getNextResultSet();
+                } catch (SQLException ex) {
+                    // TODO: https://issues.apache.org/jira/browse/IGNITE-21133
+                    // implicitly silent close previous result set.
+                }
+
+                if (last != null) {
+                    while (last != null) {
+                        try {
+                            last = last.getNextResultSet();
+                        } catch (SQLException ex) {
+                            // TODO: https://issues.apache.org/jira/browse/IGNITE-21133
+                            // implicitly silent close previous result set.
+                        }
+                    }
+                }
```

Comment:
As was discussed today - let's throw error in such situation (better late than never).

```suggestion
                do {
                    lastRs = lastRs.getNextResultSet();
                } while (lastRs != null);
```

Reply:
done